### PR TITLE
Fixes handling properties and clear union in MqttClient_WaitType

### PR DESCRIPTION
**Pull request by embhorn**
This addresses issues from:
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=39149&q=label%3AProj-wolfmqtt&sort=-id
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=39123&q=label%3AProj-wolfmqtt&sort=-id
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=39082&q=label%3AProj-wolfmqtt&sort=-id
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=39062&q=label%3AProj-wolfmqtt&sort=-id

### CODE REVIEW

File: src/mqtt_client.c [hunk 1]
```diff
@@ -952,6 +957,11 @@ static int MqttClient_WaitType(MqttClient *client, void *packet_obj,
             }
         #endif
 
+            /* Clear shared union for next call */
+            if (use_packet_obj == &client->msg) {
```
Comment (dgarske) [inline src/mqtt_client.c:961]: Consider explicit cast here on pointer compare.

File: src/mqtt_client.c [hunk 2]
```diff
@@ -952,6 +957,11 @@ static int MqttClient_WaitType(MqttClient *client, void *packet_obj,
...
+                XMEMSET(&client->msg, 0, sizeof(MqttObject));
```
Comment (dgarske) [inline src/mqtt_client.c:962]: Would prefer ‘sizeof(client->msg)’

File: src/mqtt_client.c [hunk 3]
```diff
@@ -277,6 +277,32 @@ static int MqttClient_RespList_Find(MqttClient *client,
...
+#ifdef WOLFMQTT_V5
+static int Handle_Props(MqttProp* props)
+{
+    int rc = MQTT_CODE_SUCCESS;
+
+    /* If no properties, just return */
+    if (props != NULL) {
+    #ifdef WOLFMQTT_PROPERTY_CB
+        /* Check for properties set by the server */
+        if (packet_obj && client->property_cb) {
```
Comment (dgarske) [inline src/mqtt_client.c:289]: Where are you getting ‘packet_obj’?

**Review by dgarske — CHANGES_REQUESTED** [review]

File: src/mqtt_client.c (hunk 3 above)
Comment (embhorn) [reply src/mqtt_client.c:289]: Fixed. Got in a hurry testing the fuzzer test cases.

**Comment by embhorn** [comment]
Jenkins failures appear to be related to test.mosquitto.org performance

**Review by dgarske — APPROVED** [review]

**Pull request merged**